### PR TITLE
Add SolverOfflineError exception

### PR DESCRIPTION
**Pull request by randomir**

### CODE REVIEW

File: dwave/cloud/exceptions.py [hunk 1]
```diff
@@ -14,6 +14,9 @@ class SolverError(Exception):
 class SolverFailureError(SolverError):
     """An exception raised when there is a remote failure calling a solver."""
```
Comment (JoelPasvolsky) [inline dwave/cloud/exceptions.py]: Raised when calling a remote solver fails.

File: dwave/cloud/exceptions.py [hunk 2]
```diff
@@ -14,6 +14,9 @@ class SolverError(Exception):
...
 
+class SolverOfflineError(SolverError):
+    """Action attempted on an offline solver."""
```
Comment (JoelPasvolsky) [inline dwave/cloud/exceptions.py]: Raised when a requested solver is in an offline state

File: dwave/cloud/exceptions.py [hunk 3]
```diff
@@ -14,6 +14,9 @@ class SolverError(Exception):
...
+
 class SolverAuthenticationError(SolverError):
     """An exception raised when there is an authentication error."""
```
Comment (JoelPasvolsky) [inline dwave/cloud/exceptions.py]: Raised when user authentication fails.

File: dwave/cloud/client.py [hunk 4]
```diff
@@ -901,7 +901,12 @@ def _handle_problem_status(self, message, future):
                 future._set_error(CanceledFutureError())
             else:
                 # Return an error to the future object
-                future._set_error(SolverFailureError(message.get('error_message', 'An unknown error has occurred.')))
+                errmsg = message.get('error_message', 'An unknown error has occurred.')
+                if 'solver is offline' in errmsg.lower():
+                    err = SolverOfflineError(errmsg)
+                else:
+                    err = SolverFailureError(errmsg)
```
Comment (JoelPasvolsky) [inline dwave/cloud/client.py]: How come you're not checking for the other known errors, like authentication?

Comment (randomir) [reply dwave/cloud/client.py]: They are checked for before.

**Pull request merged**